#!/usr/bin/env python

# remember to run the collofing command to make this script executable
# chmod a+x cfg/script_name.py

# Name of containing package
PACKAGE = 'l2bot_examples'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Dropdown menus use enums. The output value is just an int (3rd arg of gen.const)
# General syntax:
#   gen.enum([const1, const2, ..., constN], 'EnumName')
# where constN is:
#   gen.const('Name', int_t, N, 'Name')
thresh_type = gen.enum([    gen.const('Binary',             int_t, 0, 'Binary'),
                            gen.const('BinaryInv',          int_t, 1, 'BinaryInv'),
                            gen.const('ThresholdTrunc',     int_t, 2, 'ThreshTrunc'),
                            gen.const('ThresholdToZero',    int_t, 3, 'ThreshToZero'),
                            gen.const('ThresholdToZeroInv', int_t, 4, 'ThreshToZeroInv'),
                            gen.const('OTSU',               int_t, 5, 'OTSU') ],
                            'ThresholdType')

# Integer using a dropdown menu
gen.add('thresh_type',      int_t,      0, 'thresh_type',       0,   0,   5,   edit_method=thresh_type)

# Numeric values using sliders
# General syntax is:
#   gen.add('var_name', type, 0, 'var_name', default, low, high)
gen.add('white_all_ratio',  double_t,   0, 'white_all_ratio',   0.5, 0.0, 1.0)
gen.add('thresh_value',     int_t,      0, 'thresh_value',      180, 0,   255)
gen.add('max_value',        int_t,      0, 'max_value',         255, 0,   255)

# Checkbox using boolean value
# General syntax is:
#   gen.add('var_namne', bool_t, 0, 'var_name', default)
gen.add('use_median_blur',  bool_t,     0, 'use_median_blur',   True)
gen.add('blur_amount',      int_t,      0, 'blur_amount',       3,   0,   10)

# gen.generate(PACKAGE, 'package_name', 'PascalCaseNodeName')
exit(gen.generate(PACKAGE, 'l2bot_examples', 'StopOnWhite'))
